Fix duplicate calls to ``pool.wait_closed()`` upon ``create_pool()`` exception.
